src/sentry/digests/utils.py

```diff
@@ -97,10 +104,23 @@ def get_event_from_groups_in_digest(digest: Digest) -> Iterable[Event]:
     }
 
 
-def build_custom_digest(original_digest: Digest, events: Iterable[Event]) -> Digest:
+def build_custom_digest(
+    original_digest: Digest, events: Iterable[Event], participant: RpcActor
+) -> Digest:
     """Given a digest and a set of events, filter the digest to only records that include the events."""
     user_digest: Digest = {}
+    rule_snoozes = RuleSnooze.objects.filter(
+        Q(Q(user_id=participant.id) | Q(user_id__isnull=True)), rule__in=original_digest.keys()
+    )
+
     for rule, rule_groups in original_digest.items():
+        skip = False
+        for snooze in rule_snoozes:
+            if snooze.rule == rule:
+                skip = True
+                break
```

Comment:
Nit: Now that you have all the rule snoozes in one place, you could just store them as a set of `snoozed_rule_ids`, then say `if rule.id in snoozed_rule_ids: continue`